docs/docs/09-rfcs/2023-03-08-wing-preview-environment.md

```diff
@@ -0,0 +1,364 @@
+---
+title: Wing Cloud Preview Environments
+description: This document describes an MVP for the first Wing Cloud product - Wing Cloud Preview Environments.
+---
+
+# #3696 - Wing Cloud Preview Environments
+
+- **Author(s)**: @ainvoner, @skyrpex
+- **Submission Date**: 2023-08-03
+- **Stage**: Proposal
+- **Stage Date**: 2023-08-03
+
+This document describes an MVP for the first Wing Cloud product - Wing Cloud Preview Environments.
+
+## Introduction
+
+**Preview Environments** are the first feature of Wing Cloud. It will also be the first production grade product written in Wing language.
+
+## Requirements
+
+Wing Cloud users can build and deploy Wing applications to an ephemeral environment that uses the Wing Simulator platform, and use the Wing Console UI to interact with it.
+Once a preview environment is created, tests will be executed automatically and the results will be available in the preview environment dedicated comment in GitHub.
+Each preview environment will have a unique URL which enable developers to work together and share with external customers as needed.
+
+Wing Cloud will be integrated with GitHub.
+
+### Wing Cloud User Experience
+
+#### Sign-In to Wing Cloud (#1)
+
+1. On https://wing.cloud website click on sign in.
+2. Authenticate using your GitHub account.
+3. If you were already signed in, you will see the Wing Cloud dashboard (under https://wing.cloud directly).
+
+#### Redirect to sign-in process from a Preview Environment URL (#2)
+
+1. If you try to access a preview environment URL while you are not signed in to Wing Cloud (from the PR comment for example), you will be temporarily redirected to the Wing Cloud sign-in flow.
+2. Once singed in, you will be redirected to the preview environment URL.
+
+#### Wing Cloud Dashboard for Signed-In Users (#3)
+
+1. Once signed in, you will automatically see Wing Cloud for signed-in users (under https://wing.cloud directly).
+2. You can see and manage all your projects and teams.
+3. You will see all preview environments associated with your personal account and teams you are member in.
+4. In order to see a preview environment, just click on the preview environment URL from the list of all your preview environments.
+5. For each preview environment, you can see the deployment logs and the test results and logs.
+
+#### Create a new personal Project (#4)
+
+1. To create a new project, click on "New Project" under "Personal Account".
+2. Choose between the GitHub repositories that you have access to.
+3. Define your secret values.
+4. Make sure there's a `main.w` file in the root of the repository.
+5. Click on "Deploy".
+6. You'll be redirected to your new project's page (https://wing.cloud/-user/-project).
```

Comment:
Personal account is mentioned here and also `-user` in the path. 
Did you guys have the organization vs personal account discussion? 

From my experience this is a complex topic, especially when you start to think about analytics, permission, multi organizations and migrating projects. 

I suggest we spend some time together on this topic to make sure we start as simple as possible.